src/Symfony/Component/DomCrawler/Crawler.php

```diff
@@ -593,6 +593,33 @@ public function html()
     }
 
     /**
+     * Evaluates an XPath expression.
+     *
+     * Since an XPath expression might evaluate to either a simple type or a \DOMDoneList,
+     * this method will return either an array of simple types or a new Crawler instance.
+     *
+     * @param string $xpath An XPath expression
+     *
+     * @return array|Crawler An array of evaluation results or a new Crawler instance
+     */
+    public function evaluate($xpath)
+    {
+        $data = array();
+        $prefixes = $this->findNamespacePrefixes($xpath);
+        $domxpath = $this->createDOMXPath($this->document, $prefixes);
```

Comment:
I would throw a LogicException before that in case ``$this->document`` is null, saying you cannot evaluate an expression on an uninitialized crawler.